cmip6_downscaling/methods/bcsd.py

```diff
@@ -0,0 +1,249 @@
+        chunk_dims=('lat', 'lon'),
+        connection_string=connection_string,
+        max_mem='1GB',
+    )
+
+    return y_rechunked_path, X_train_rechunked_path, X_predict_rechunked_path
+
+
+def fit_and_predict(
+    X_train_path: str,
+    y_path: str,
+    X_predict_path: str,
+    bias_corrected_path: str,
+    dim: str = "time",
+    variable: str = "tasmax",
+):
+    """Fit bcsd model on prepared CMIP data with obs at corresponding spatial scale.
+    Then predict for a set of CMIP data (likely future).
+
+    Parameters
+    ----------
+    X_train_path : str
+        Path to GCM training data chunked along space
+    y_path : str
+        Path to obs training data chunked along space
+    X_predict_path : str
+        Path to GCM prediction data chunked along space.
+    bias_corrected_path : str
+        Path to final bias corrected data (matches dims of X_predict).
+    dim : str, optional
+        dimension on which you want to do the modelling, by default "time"
+    variable : str, optional
+        variable you're modelling, by default "tasmax"
+
+    Returns
+    -------
+    bias_corrected_path : str
+        Path to where coarse bias-corrected data is
+    """
+    y, X_train, X_predict = load_paths([y_path, X_train_path, X_predict_path])
+    if variable in ABSOLUTE_VARS:
+        bcsd_model = BcAbsolute(return_anoms=False)
+    elif variable in RELATIVE_VARS:
+        bcsd_model = BcRelative(return_anoms=False)
+    pointwise_model = PointWiseDownscaler(model=bcsd_model, dim=dim)
+    pointwise_model.fit(X_train[variable], y[variable])
+    bias_corrected = pointwise_model.predict(X_predict[variable])
+    write_dataset(bias_corrected.to_dataset(name=variable), bias_corrected_path)
+    return bias_corrected_path
+
+
+def postprocess_bcsd(
+    y_predict_path: str,
+    spatial_anomalies_path: str,
+    final_out_path: str,
+    variable: str,
+    connection_string: str,
+):
+    """Downscale the bias-corrected data by interpolating and then
+    adding the spatial anomalies back in.
+
+    Parameters
+    ----------
+    y_predict_path : str
+        location where bias-corrected data is.
+    spatial_anomalies_path : str
+        location where spatial anomalies are
+    final_out_path : str
+        location to write final BCSD data
+    variable : str
+        variable you're working on
+    connection_string : str
+        Connection string to give you read/write access to the out buckets specified above
+
+    Returns
+    -------
+    final_out_path : str
+        location where final BCSD data was written
+    """
+    y_predict, spatial_anomalies = load_paths([y_predict_path, spatial_anomalies_path])
+
+    # TODO: test - create sample input, run it through rechunk/regridder and then
+    # assert that it looks like i want it to
+    y_predict_fine = regrid_dataset(y_predict, spatial_anomalies, variable, connection_string)
+    bcsd_results = y_predict_fine.groupby("time.month") + spatial_anomalies
```

Comment:
nit/question: Will "time.month" change in future runs? Should this be an arg or is it fine as is.

Reply:
good question! this will always be `time.month`. this method works by calculating seasonal averages (aka one value for every month of the year- 12 values). if we were to switch to daily bcsd we'd have to change a lot of other things, so i think this is fine as-is.